infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -157,16 +164,14 @@ func NewTFBlueprintTest(t testing.TB, opts ...tftOption) *TFBlueprintTest {
 	if tft.setupDir != "" {
 		tft.logger.Logf(tft.t, "Loading env vars from setup %s", tft.setupDir)
 		loadTFEnvVar(tft.tfEnvVars, tft.getTFOutputsAsInputs(terraform.OutputAll(tft.t, &terraform.Options{TerraformDir: tft.setupDir, Logger: tft.logger})))
-		// setup credentials if explicit sa_key output from setup
-		credsEnc, exists := tft.tfEnvVars[fmt.Sprintf("TF_VAR_%s", setupKeyOutputName)]
-		if !exists {
-			tft.logger.Logf(tft.t, "Unable to find %s output from setup, skipping credential activation", setupKeyOutputName)
+		if tft.saKey != "" {
+			tft.loadSetupSAKey(tft.saKey)
 		} else {
```

Comment:
This is nested within `if tft.setupDir != ""{..}` so if there is no `setupDir` control wont reach here. We could check and activate SA before this if ` tft.saKey != ""` so it works even without setup.